### PR TITLE
Add command for sync forks repositories

### PR DESCRIPTION
**Pull request by DAGpro**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #95

### CODE REVIEW

File: src/App/Command/Github/SyncUpstreamRepositoriesCommand.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+final class SyncUpstreamRepositoriesCommand extends PackageCommand
+{
+    protected static $defaultName = 'github/sync';
+    protected static $defaultDescription = 'Sync forks with upstream repositories';
+
+    protected function configure()
+    {
+        $this
+            ->setAliases(['sync'])
+            ->setDescription('Sync forks with upstream repositories');
```
Comment (vjik) [inline src/App/Command/Github/SyncUpstreamRepositoriesCommand.php:23]: `setDescription()` not need when use `$defaultDescription`.

File: src/App/Command/Github/SyncUpstreamRepositoriesCommand.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    '<error>Check if the nickname of the owner and the name of the repository are correct</error>' . PHP_EOL,
+                ]
+            );
+        }
+    }
+
+    public function getGithubApiRepository(): Repo
+    {
+        $client = new Client();
+        $client->authenticate($this->getToken(), null, Client::AUTH_ACCESS_TOKEN);
```
Comment (vjik) [inline src/App/Command/Github/SyncUpstreamRepositoriesCommand.php:58]: `AUTH_ACCESS_TOKEN` is depricated, use `AuthMethod::ACCESS_TOKEN`.

File: src/App/Command/Github/SyncUpstreamRepositoriesCommand.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            if (isset($repoMatches[1])) {
+                $repository->syncUpstream($package->getVendor(), $repoMatches[1]);
+                $this->getIO()->write("<success>Repository successfully synced: {$package->getName()}</success>" . PHP_EOL);
+            } else {
+                $this->getIO()->error(
+                    [
+                        "<error>Error when sync a repository {$package->getName()}</error>",
+                        '<error>Check if the nickname of the owner and the name of the repository are correct</error>' . PHP_EOL,
+                    ]
+                );
```
Comment (vjik) [inline src/App/Command/Github/SyncUpstreamRepositoriesCommand.php:43]: This code is very similar to the code belowю May be move this to separate private method?

File: README.md [hunk 4]
```diff
@@ -245,6 +245,7 @@ Remember that **yii-dev-tool** contains many other commands:
 | git/status                | status, s | Show git status of packages.                                                 |
 | git/pr/create             | pr        | Create a pull request at GitHub.                                             |
 | github/forks              | forks     | Create forks of repositories                                                 |
+| github/sync               | sync      | Sync forks upstream repositories                                             |
```
Comment (samdark) [inline README.md:248]:
```suggestion
| github/sync               | sync      | Sync forks from upstream repositories                                             |
```

File: src/App/Command/Github/SyncUpstreamRepositoriesCommand.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                $this->errorMessage($package->getName(), $e->getMessage())
+            );
+        }
+    }
+
+    private function errorMessage($packageName, $message = ''): array
+    {
+        $error = $message ? "{$packageName}: {$message}" : $packageName;
+        return [
+            "<error>Error when sync a repository $error </error>",
```
Comment (samdark) [inline src/App/Command/Github/SyncUpstreamRepositoriesCommand.php:53]:
```suggestion
            "<error>Error when syncing a repository $error </error>",
```

File: src/App/Command/Github/SyncUpstreamRepositoriesCommand.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        return new class ($client) extends Repo {
+            public function mergeUpstream($username, $repository, $branchName = null)
+            {
+                return $this->post(
+                    '/repos/' . rawurlencode($username) . '/' . rawurlencode($repository) . '/merge-upstream',
+                    ['branch' => $branchName ?? 'main']
+                );
+            }
+        };
+        // return (new Repo($client));
```
Comment (samdark) [inline src/App/Command/Github/SyncUpstreamRepositoriesCommand.php:73]: Remove?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thanks!

**Pull request merged**